Userland/Libraries/LibWeb/HTML/WindowOrWorkerGlobalScope.cpp

```diff
@@ -309,9 +310,7 @@ i32 WindowOrWorkerGlobalScopeMixin::run_timer_initialization_steps(TimerHandler
     };
 
     // 12. Run steps after a timeout given global, "setTimeout/setInterval", timeout, completionStep, and id.
-    auto timer = Timer::create(this_impl(), timeout, move(completion_step), id);
-    m_timers.set(id, timer);
-    timer->start();
+    run_steps_after_a_timeout_impl(timeout, move(completion_step), id);
```

Comment:
This bit really belongs in its own commit

Reply:
I've split this into it's own commit and added (mostly FIXME) spec comments to the function.